pkg/scaffold/v1/resource/resource.go

```diff
@@ -79,7 +80,10 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf(
 			"version must match ^v\\d+(alpha\\d+|beta\\d+)?$ (was %s)", r.Version)
 	}
-	if r.Kind != flect.Pascalize(r.Kind) {
+
+	validationErrors := validation.IsDNS1035Label(strings.ToLower(r.Kind))
```

Comment:
See here that the import is not required. In the PR https://github.com/kubernetes-sigs/kubebuilder/pull/1048/files#diff-ccdb7c3b4336337dc2e653f1fbd88c7dR71 a similar validation from apimachinery was added in the project. There was decide do not do the import. So, I'd recommend you do the same.